### PR TITLE
Topic recognition - Variation Autoencoder

### PR DESCRIPTION
**Pull request by qykcq**
My work can be found at PatternFlow/recognition/VAE.
VAE.py defines the network structure. 
driver.py contains the runnable training script.
To run the script, use the command:
> python driver.py arg1 arg2
where arg1 is the folder that contains the training images, and arg2 is the folder that contains the test images.
PatternFlow/recognition/VAE/images contains the plots. 

Author:
Yunke Qu. Student number: 44631604

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Some commit messages should be more specific (e.g. what "bug" did you fix)
- Code valid for Hard Difficulty
- Valid results

Other Feedback
- gitignore `.idea`
- Please follow python file  and folder naming convention e.g. `lorem_ipsum_123_abc.py` (vae.py)
- Good job on the VAE. Quick tip: the variation layers can be imported from `tensorflow-probability` package.

**Comment by rexsw** [comment]
Code looks good for hard difficulty

For next time;

Comments such as 
      # the encoder
        self.encoder = self.define_encoder(latent_dimension, kernel_size, strides)
Add nothing to the readability of the code and should be avoided. Overall comments were well done

If you try something in python and it fails it's a good idea to print/throw an error message

File: recognition/VAE/VAE.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/VAE/VAE.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Fix conflict with the .idea files. Otherwise I can't merge.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**